tt_metal/llrt/tt_cluster.cpp

```diff
@@ -921,46 +921,47 @@ void Cluster::set_internal_routing_info_for_ethernet_cores(bool enable_internal_
     for (const auto &[assoc_mmio_device, devices] : this->devices_grouped_by_assoc_mmio_device_) {
+        mmio_devices.emplace_back(assoc_mmio_device);
         for (const auto &chip_id : devices) {
+            non_mmio_devices.emplace_back(chip_id);
+        }
+    }
+
+    if (enable_internal_routing) {
+        for (const auto &chip_id : non_mmio_devices) {
             for (const auto &[eth_core, routing_info] : this->device_eth_routing_info_.at(chip_id)) {
                 tt_cxy_pair eth_phys_core(chip_id, ethernet_core_from_logical_core(chip_id, eth_core));
-                if (chip_id == assoc_mmio_device and not enable_internal_routing) {
-                    // Disable internal ethernet routing for mmio devices
-                    write_core(
-                        (void *)&routing_info_disabled,
-                        sizeof(routing_info_t),
-                        eth_phys_core,
-                        routing_info_addr,
-                        false);
-                } else if (chip_id != assoc_mmio_device and enable_internal_routing) {
-                    // Enable internal ethernet routing for non-mmio devices
-                    write_core(
-                        (void *)&routing_info_enabled, sizeof(routing_info_t), eth_phys_core, routing_info_addr, false);
-
-                } else {
-                    continue;
-                }
+                // Enable internal ethernet routing for non-mmio devices
+                write_core(
+                    (void *)&routing_info_enabled, sizeof(routing_info_t), eth_phys_core, routing_info_addr, false);
             }
         }
-        for (const auto &chip_id : devices) {
+        for (const auto &chip_id : mmio_devices) {
             for (const auto &[eth_core, routing_info] : this->device_eth_routing_info_.at(chip_id)) {
                 tt_cxy_pair eth_phys_core(chip_id, ethernet_core_from_logical_core(chip_id, eth_core));
-                if (chip_id != assoc_mmio_device and not enable_internal_routing) {
-                    // Disable internal ethernet routing for non-mmio devices
-                    write_core(
-                        (void *)&routing_info_disabled,
-                        sizeof(routing_info_t),
-                        eth_phys_core,
-                        routing_info_addr,
-                        false);
-                } else if (chip_id == assoc_mmio_device and enable_internal_routing) {
-                    // Enable internal ethernet routing for mmio devices
-                    write_core(
-                        (void *)&routing_info_enabled, sizeof(routing_info_t), eth_phys_core, routing_info_addr, false);
-                } else {
-                    continue;
-                }
+                // Enable internal ethernet routing for mmio devices
+                write_core(
+                    (void *)&routing_info_enabled, sizeof(routing_info_t), eth_phys_core, routing_info_addr, false);
```

Comment:
maybe move the `routing_info_enabled`, `routing_info_disabled` inside the if/else scope?